x-pack/plugins/security_solution/public/notes/store/notes.slice.ts

```diff
@@ -129,9 +129,21 @@ export const createNote = createAsyncThunk<NormalizedEntity<Note>, { note: BareN
 
 export const deleteNotes = createAsyncThunk<string[], { ids: string[] }, {}>(
   'notes/deleteNotes',
-  async (args) => {
+  async (args, { dispatch, getState }) => {
     const { ids } = args;
     await deleteNotesApi(ids);
+    const state = getState() as State;
+    const { search, pagination, sort } = state.notes;
+    dispatch(
```

Comment:
actually it's even worth than that, when deleting a note in the flyout, there is a pretty big chance we're nuking everything. We're using `setAll` when setting the notes once the `fetchNotes` succeeds, so we're basically clearing the entire slice and adding the new notes. There is no guarantee that these notes will be ones for the document id of the opened flyout

https://github.com/elastic/kibana/assets/17276605/c5638166-8c1d-4b1b-a5f5-de01f3c1e073

Reply:
I see a few ideas on how to solve this:

- have a boolean `refetch` that controls if we want to refetch or not, like so
```typescript
export const deleteNotes = createAsyncThunk<string[], { ids: string[]; refetch?: boolean }, {}>(
  'notes/deleteNotes',
  async (args, { dispatch, getState }) => {
    const { ids, refetch } = args;
    await deleteNotesApi(ids);

    if (refetch) {
      const state = getState() as State;
      const { search, pagination, sort } = state.notes;
      dispatch(
        fetchNotes({
          page: pagination.page,
          perPage: pagination.perPage,
          sortField: sort.field,
          sortOrder: sort.direction,
          filter: '',
          search,
        })
      );
    }

    return ids;
  }
);
```
This would work now but if we ever implement pagination in the flyout we'll have the problem of wanting to use a different refetch function


- have 2 different thunks, like so
```typescript
export const deleteNotes = createAsyncThunk<string[], { ids: string[] }, {}>(
  'notes/deleteNotes',
  async (args) => {
    const { ids } = args;
    await deleteNotesApi(ids);
    return ids;
  }
);

export const deleteNotesAndRefetch = createAsyncThunk<string[], { ids: string[] }, {}>(
  'notes/deleteNotes',
  async (args, { dispatch, getState }) => {
    const { ids } = args;
    await deleteNotesApi(ids);
    const state = getState() as State;
    const { search, pagination, sort } = state.notes;
    dispatch(
      fetchNotes({
        page: pagination.page,
        perPage: pagination.perPage,
        sortField: sort.field,
        sortOrder: sort.direction,
        filter: '',
        search,
      })
    );
    return ids;
  }
);
```


- update the `total` value in the slice directly in the `deleteNotes.fulfilled`, like so
```typescript
.addCase(deleteNotes.fulfilled, (state, action) => {
        notesAdapter.removeMany(state, action.payload);
        state.status.deleteNotes = ReqStatus.Succeeded;
        state.pendingDeleteIds = state.pendingDeleteIds.filter(
          (value) => !action.payload.includes(value)
        );
        state.pagination.total -= 1;
      })
```
But this will no reset the pages properly, so we will still have issue if we delete a bunch of notes...


- do the logic of await for the thunk to finish somewhere else (in the component?)

Reply:
I went with the first one. I have a feeling pagination in the flyout will be more of an infinite scroll/append only pagination, and so will use different thunks anyway. Thanks for the suggestions!